chainerx_cc/chainerx/float16.h

```diff
@@ -0,0 +1,40 @@
+#pragma once
+
+#include <cstdint>
+
+namespace chainerx {
+
+struct Half {
+public:
+    Half() {}
+    Half(const Half& v) : data_(v.data_) {}
+    explicit Half(float v);
```

Comment:
Should this be `explicit`?

Reply:
Travis test failed with cpplint: `Single-parameter constructors should be marked explicit.`, so I'll make all such ctors explicit.

Reply:
Discussed offline and decided to be conservative about implicit casts.